modules/migrations/gitea.go

```diff
@@ -0,0 +1,403 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Copyright 2018 Jonas Franz. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/migrations/base"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+
+	gouuid "github.com/satori/go.uuid"
+)
+
+var (
+	_ base.Uploader = &GiteaLocalUploader{}
+)
+
+// GiteaLocalUploader implements an Uploader to gitea sites
+type GiteaLocalUploader struct {
```

Comment:
@kolaente there are uploaders and downloaders here. `GiteaLocalUploader` will save data via gitea's local operations. Currently we have `GithubV3Downloader`, in future, we could implement GitLab/Gitea/Gogs/... downloaders. Currently we could have `GithubV3Uploader`/GitLab so that we can migrate gitea's data to github.

Reply:
So the flow is `GithubV3Uploader` -> `GiteaLocalUploader`?

Reply:
Current flow is `GithubV3Downloader` - > `GiteaLocalUploader`. Always from a downloader interface implementation to an uploader interface implementation.